lib/rbac/filterer.rb

```diff
@@ -417,17 +402,33 @@ def scope_to_tenant(scope, user, miq_group)
       tenant_id_clause ? scope.where(tenant_id_clause) : scope
     end
 
+    ##
+    # Main scoping method
+    #
     def scope_targets(klass, scope, rbac_filters, user, miq_group)
+      # Results are scoped by tenant if the TenancyMixin is included in the class,
+      # with a few manual exceptions (User, Tenant). Note that the classes in
+      # TENANT_ACCESS_STRATEGY are a consolidated list of them.
       if klass.respond_to?(:scope_by_tenant?) && klass.scope_by_tenant?
```

Comment:
> If that's the case, should this check be klass < TenancyMixin && klass.scope_by_tenant? ? Not sure that's better but it matches your comment better.

As @chrisarcand says, `User` and `Tenant` are special and implement this method `scope_by_tenant?` on their own outside of the tenancy mixin.  Maybe we can clarify this comment then.